src/process/mod.rs

```diff
@@ -510,15 +557,125 @@ impl MMapPath {
     }
 }
 
-/// Represents an entry in a `/proc/<pid>/maps` file.
+/// Represents all entries in a `/proc/<pid>/maps` or `/proc/<pid>/smaps` file.
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde1", derive(Serialize, Deserialize))]
+pub struct MemoryMaps {
```

Comment:
It might make sense to declare this struct `#[non_exhaustive]` so we can add new fields in the future in a backwards-compatible way